binary-extensions: add cibuildwheel and multibuild (#910)
pypa/packaging.python.org: source/guides/packaging-binary-extensions.rst

Building binary extensions
==========================

Building extensions for multiple platforms

> -------------------------

Reviewer: This underline needs to be fixed tho. :)

Author: Too late. ;)

Building binary extensions
==========================

> Building extensions in CI

Reviewer 2: Readers might not know what CI is, and I think focusing on the desired outcome is clearer:

```suggestion
Building extensions for multiple platforms
```

-------------------------

> If you plan to distribute your modules, you should provide builds for all the
> platforms you intend to support. For most extensions, this is at least one
> module per Python version times the number of OS and archetectures you support.

Reviewer 2: I _think_ this is more accurate, based on my understanding:

```suggestion
If you plan to distribute your extension, you should provide :term:`wheels <Wheel>` for all the
platforms you intend to support. For most extensions, this is at least one
package per Python version times the number of OS and architectures you support.
```

These are usually built on continous integration (CI) systems. There are tools
to help you build highly redistributable binaries from CI; these include

> `cibuildwheel <https://cibuildwheel.readthedocs.io>`__ and
> `multibuild <https://github.com/matthew-brett/multibuild>`__.

Reviewer 2: I think the pattern has been to link to the [project summaries](https://github.com/pypa/packaging.python.org/blob/main/source/key_projects.rst):

```suggestion
:ref:`cibuildwheel` and :ref:`multibuild`.
```

==========================

Building extensions for multiple platforms
------------------------------------------

If you plan to distribute your extension, you should provide :term:`wheels

> <Wheel>` for all the platforms you intend to support. For most extensions, this

Reviewer: I don't think splitting a directive across multiple lines works, but hey, I'd be willing to be surprised. :)

Author: Ahh, reflowed that and didn't watch for directives. We can see, I guess. :)

Author: Nope.
